contentstore/tests/tests_server.py

```diff
@@ -0,0 +1,456 @@
+        resp = self.api.handle_request(
+            self.req_class('GET', path, None, self.headers))
+        return resp.code, resp.data
+
+    def post(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('POST', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def put(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('PUT', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def patch(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('PATCH', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def delete(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('DELETE', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+    def options(self, path, data=None, format=None, content_type=None):
+        data, content_type = self._encode_data(data, format, content_type)
+        self.headers["Content-Type"] = content_type
+        resp = self.api.handle_request(self.req_class('OPTIONS', path, data,
+                                                      self.headers))
+        return FakeResponse(resp.code, resp.data)
+
+
+class TestFakeContentStore(TestCase, ContentStoreApiTestMixin):
+
+    def setUp(self):
+        try:
+            from verified_fake.fake_contentstore import (Request,
+                                                         FakeContentStoreApi)
+        except ImportError as err:
+            if "verified_fake" not in err.args[0]:
+                raise
+            raise ImportError(" ".join([
+                err.args[0],
+                "(install from pypi or the 'verified_fake' directory)"]))
+
+        self.req_class = Request
+        self.api_class = FakeContentStoreApi
+        self.api = self.api_class("", "token-1", {}, {})
+        self.api.messagesets.endpoint_data = {}
+        self.api.schedules.endpoint_data = {}
+        self.api.messages.endpoint_data = {}
+        self.api.binary_contents.endpoint_data = {}
```

Comment:
This seems like a lot of setup code. We probably want the user of the fake api to just be able to `api = FakeContentStoreApi(..)`. What is the reason for setting the endpoint data to empty dictionaries manually?